### DISCUSSION TITLE
Locked and unlocked by.

### DISCUSSION THREAD

**Discussion by pmkone** [tsightler/ring-mqtt#885] [Feature Requests]
I realize there are disarmedby, lastarmedby, attributes. I am eternally grateful for those. Any chance of getting an unlocked and locked by for the locks? Similar to which user armed it last. Would be great to run automations based on who unlocked a door.

**Comment by tsightler** [comment]
I would say almost certainly not unless somebody else wants to code it up.  Implementing it security panel was a pain to get right as this data is only available in realtime as part of the impulse data, and this isn't currently included in state updates from ring-client-api, so you have to monitor and filter our state updates for each device.  Technically, it's also available in the event data, but I'm not too excited about delaying state updates to do event and user lookups, which might fail, etc.

Realistically, if you want to do that level of advanced automations with locks, I'd suggest you're better off paring them them directly to your automation tool vs gatewaying them them with some proprietary Z-wave hub, then you can automate however you want.
